src/main/java/org/opensearch/plugin/insights/core/service/grouper/QueryGrouper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.service.grouper;
+
+import org.opensearch.plugin.insights.core.service.store.PriorityQueueTopQueriesStore;
+import org.opensearch.plugin.insights.rules.model.GroupingType;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+
+/**
+ * Interface for grouping search queries based on grouping type for the metric type.
+ */
+public interface QueryGrouper {
+
+    /**
+     * Add query to the group based on the GroupType setting.
+     * @param searchQueryRecord record to be added
+     * @return the aggregate search query record representing the group
+     */
+    SearchQueryRecord addQueryToGroup(SearchQueryRecord searchQueryRecord);
```

Comment:
[NIT] I guess we can just call it ```add```? As adding queryRecord is self implied within interface